components/modal/src/ModalHeader.js

```diff
@@ -0,0 +1,48 @@
+/** @jsx jsx */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { jsx, useTheme } from '@westpac/core';
+import { CloseIcon } from '@westpac/icon';
+import { Title } from './styled';
+
+// ==============================
+// Component
+// ==============================
+export const ModalHeader = ({ id, onClose, children, ...props }) => {
+	const { modal, colors } = useTheme();
+
+	return (
+		<div
+			css={{
+				display: 'flex',
+				justifyContent: 'space-between',
+				alignItems: 'center',
+				overflow: 'hidden',
+				borderBottom: `${modal.header.borderWidth} solid ${modal.header.borderColor}`,
+				padding: modal.header.padding,
+			}}
+			{...props}
+		>
+			<Title id={id}>{children}</Title>
+			<button
+				css={{ border: 'none', padding: 0, '&:hover svg': { opacity: 0.5 } }}
+				onClick={onClose}
+			>
+				<CloseIcon size="small" color={colors.text} />
+			</button>
```

Comment:
can we use our `Button`?

Reply:
^ will also help address the button click area. This button is way too small

Reply:
Fixed the button sizing. 

I still believe that it is unnecessary to use the GEL button for simple things like this. It would be good to get ThinkMill's opinion on this. If they think we should use the GEL button then I will change to use it otherwise I will leave it as is for now.

Reply:
Hmm

Reply:
Spoke to Dom, he said that we should be reusing our components and be "atomic". Build small blocks and reuse them throughout. Any file size and memory usage etc concerns negligible 🤷‍♂️

I think we should implement our Button and mark this as a thing to discuss in depth.

Reply:
We need to introduce a prop for button to be able to take in icon color if we are going to do it this way.

Reply:
Update to new button with the 'color: inherit' so this isnt an issue anymore